src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+ 
+        // verify that no new address was created
+        ListAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Contact__c from Address__c];
+        system.assertEquals(2, ListAddr.size());
+        for (Address__c addr : ListAddr) {
+            system.assert(addr.MailingStreet__c.equals('direct street  edit'));
+        }                   
+    }
+
+    /*********************************************************************************************************
+    @description Update Contacts' Mailing address to cause an update of the default Address object.  
+    verify:
+        existing Default Address updated
+        Contact get updated Address
+    **********************************************************************************************************/            
+    @isTest
+    public static void updateAddrExisting() {
+        configSettings();
+        
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+
+        // modify some of the Contact addresses directly
+        // NOTE: we only modify 1 field, so it will be treated as an update to an existing address!
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = 'Direct Street Edit';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts;
+
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingState = 'Washington';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts;
+
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = ' direct street  edit ';    // whitespace and casing should not count as real edit.
+            contactsAddrs.contacts[i].MailingState = 'Oregon';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.contacts;
+        Test.stopTest();
+        
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assert(contact.MailingStreet.contains('direct street  edit'));
+            system.assert(contact.MailingState.contains('Oregon')); 
+            system.assertEquals(false, contact.is_Address_Override__c);
+            system.assertNotEquals(null, contact.Current_Address__c);
+        }
+ 
+        // verify that no new address was created
+        List<Address__c> ListAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Contact__c from Address__c];
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. I suggest to introduce a `WHERE` clause or cap the max number of records returned adding a `LIMIT` clause to the query.